### DISCUSSION TITLE
How to use proxies

### DISCUSSION THREAD

**Discussion by RobertAzovski** [thedemons/opentele#104] [General]
What is correct way to obtain TelegramClient instance from TDATA but to be sure it uses proxies when connecting to an account ? Because now after several days of usage I still see that my proxy traffic wasnot decreased and the same 0.77 GB, (after several days of usage Opentele !). So I suspect that there is a some bug or some another problem. May be because it is rotating proxy ? The proxy has where it is host it is not common ip address but HOST name like a string. Does Opentele/Telethon sopport such proxies ?

Example one. This is how I use Opentele now.
`client = await tdesk.ToTelethon(session=None, flag=UseCurrentSession, proxy=proxy)`

Example two. Never tried this way: `client = await TelegramClient.FromTDesktop(tdesk, session=None, flag=UseCurrentSession, proxy=proxy)`

What do I do wrong ?

**Comment by RobertAzovski** [comment]
I fixed the problem.

1. I had to install python-socks library to have fullsupport of proxies in Telethon `pip install python-socks[asyncio]`
2. I used wrong format of proxies tuple. So it is better to use a dict format as it is in Telethon docs.

`proxy = {
            'proxy_type': 'http', # (mandatory) protocol to use (see above)
            'addr': PROXY_HOST,      # (mandatory) proxy IP address
            'port': PROXY_PORT,           # (mandatory) proxy port number
            'username': PROXY_USER,      # (optional) username if the proxy requires auth
            'password': PROXY_PASSWORD,      # (optional) password if the proxy requires auth
            'rdns': True            # (optional) whether to use remote or local resolve, default remote
        }`

**Reply by justclassic552** [reply]
Hi! Have you used this in FromTDesktop or ToTelethon?

**Reply by RobertAzovski** [reply]
Hi. I tried both and I don't see any difference
